systemuser.sh

```diff
@@ -100,55 +100,7 @@ fi
-                        && echo "c.InteractiveShellApp.extensions.append('\''sparkmonitor.kernelextension'\'')" >>  $KERNEL_PROFILEPATH \
-                        && if [[ $SPARK_CLUSTER_NAME ]]; \
-                           then \
-                             echo "Configuring environment for Spark cluster: $SPARK_CLUSTER_NAME"; \
-                             source $SPARK_CONFIG_SCRIPT $SPARK_CLUSTER_NAME; \
-                             export SPARK_LOCAL_IP=`hostname -i`; \
-                             NBCONFIG=$SCRATCH_HOME/.jupyter/nbconfig ; \
-                             mkdir -p $NBCONFIG ; \
-                             echo "{
-                               \"load_extensions\": {
-                                 \"sparkconnector/extension\": true
-                               }
-                             }" > $NBCONFIG/notebook.json ; \
-                             echo "c.InteractiveShellApp.extensions.append('\''sparkconnector.connector'\'')" >>  $KERNEL_PROFILEPATH; \
-                             echo "Completed Spark Configuration" ; \
-                           fi \
-                        && export JUPYTER_DATA_DIR=$LCG_VIEW/share/jupyter \
-                        && export TMP_SCRIPT=`mktemp` \
-                        && if [[ $USER_ENV_SCRIPT && -f `eval echo $USER_ENV_SCRIPT` ]]; \
-                           then \
-                             echo "Found user script: $USER_ENV_SCRIPT"; \
-                             export TMP_SCRIPT=`mktemp`; \
-                             cat `eval echo $USER_ENV_SCRIPT` > $TMP_SCRIPT; \
-                             source $TMP_SCRIPT; \
-                           else \
-                             echo "Cannot find user script: $USER_ENV_SCRIPT"; \
-                           fi \
-                        && cd $KERNEL_DIR \
-                        && python -E -c "import os; kdirs = os.listdir(\"./\"); \
-                           kfile_names = [\"%s/kernel.json\" %kdir for kdir in kdirs]; \
-                           kfile_contents = [open(kfile_name).read() for kfile_name in kfile_names]; \
-                           exec(\"def addEnv(dtext): d=eval(dtext); d[\\\"env\\\"]=dict(os.environ); return d\"); \
-                           kfile_contents_mod = map(addEnv, kfile_contents); \
-                           import json; \
-                           print kfile_contents_mod; \
-                           map(lambda d: open(d[0],\"w\").write(json.dumps(d[1])), zip(kfile_names,kfile_contents_mod)); \
-                           termEnvFile = open(\"$SWAN_ENV_FILE\", \"w\"); \
-                           [termEnvFile.write(\"export %s=\\\"%s\\\"\\n\" % (key, val)) if key != \"SUDO_COMMAND\" else None for key, val in dict(os.environ).iteritems()];" \
-                        && printf "alias python=\"$(which python$PYVERSION)\"\n" >> $SWAN_ENV_FILE '
+sudo -E -u $USER sh /srv/singleuser/userconfig.sh
```

Comment:
Should we do a check of how the execution of the user script went, and if the process ended with error code != 0 we error out? Something like "Error running user environment script"

Reply:
Just to put a log message, or should we do anything else with that information?
If something breaks in that script we should see it already in the logs.

Reply:
Yes, but I want also a message from us saying that it was the user environment script failing, so that debugging is faster (we know where the issue is)

Reply:
done